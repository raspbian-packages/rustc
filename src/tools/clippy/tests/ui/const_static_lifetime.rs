#[derive(Debug)]
struct Foo {}

const VAR_ONE: &'static str = "Test constant #1"; // ERROR Consider removing 'static.

const VAR_TWO: &str = "Test constant #2"; // This line should not raise a warning.

const VAR_THREE: &[&'static str] = &["one", "two"]; // ERROR Consider removing 'static

const VAR_FOUR: (&str, (&str, &'static str), &'static str) = ("on", ("th", "th"), "on"); // ERROR Consider removing 'static

const VAR_FIVE: &'static [&[&'static str]] = &[&["test"], &["other one"]]; // ERROR Consider removing 'static

const VAR_SIX: &'static u8 = &5;

const VAR_SEVEN: &[&(&str, &'static [&'static str])] = &[&("one", &["other one"])];

const VAR_HEIGHT: &'static Foo = &Foo {};

const VAR_SLICE: &'static [u8] = b"Test constant #1"; // ERROR Consider removing 'static.

const VAR_TUPLE: &'static (u8, u8) = &(1, 2); // ERROR Consider removing 'static.

const VAR_ARRAY: &'static [u8; 1] = b"T"; // ERROR Consider removing 'static.

fn main() {
    let false_positive: &'static str = "test";
    println!("{}", VAR_ONE);
    println!("{}", VAR_TWO);
    println!("{:?}", VAR_THREE);
    println!("{:?}", VAR_FOUR);
    println!("{:?}", VAR_FIVE);
    println!("{:?}", VAR_SIX);
    println!("{:?}", VAR_SEVEN);
    println!("{:?}", VAR_HEIGHT);
    println!("{}", false_positive);
}

trait Bar {
    const TRAIT_VAR: &'static str;
}

impl Foo {
    const IMPL_VAR: &'static str = "var";
}

impl Bar for Foo {
    const TRAIT_VAR: &'static str = "foo";
}
