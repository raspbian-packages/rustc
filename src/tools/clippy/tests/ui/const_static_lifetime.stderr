error: Constants have by default a `'static` lifetime
 --> $DIR/const_static_lifetime.rs:4:17
  |
4 | const VAR_ONE: &'static str = "Test constant #1"; // ERROR Consider removing 'static.
  |                -^^^^^^^---- help: consider removing `'static`: `&str`
  |
  = note: `-D const-static-lifetime` implied by `-D warnings`

error: Constants have by default a `'static` lifetime
 --> $DIR/const_static_lifetime.rs:8:21
  |
8 | const VAR_THREE: &[&'static str] = &["one", "two"]; // ERROR Consider removing 'static
  |                    -^^^^^^^---- help: consider removing `'static`: `&str`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:10:32
   |
10 | const VAR_FOUR: (&str, (&str, &'static str), &'static str) = ("on", ("th", "th"), "on"); // ERROR Consider removing 'static
   |                               -^^^^^^^---- help: consider removing `'static`: `&str`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:10:47
   |
10 | const VAR_FOUR: (&str, (&str, &'static str), &'static str) = ("on", ("th", "th"), "on"); // ERROR Consider removing 'static
   |                                              -^^^^^^^---- help: consider removing `'static`: `&str`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:12:18
   |
12 | const VAR_FIVE: &'static [&[&'static str]] = &[&["test"], &["other one"]]; // ERROR Consider removing 'static
   |                 -^^^^^^^------------------ help: consider removing `'static`: `&[&[&'static str]]`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:12:30
   |
12 | const VAR_FIVE: &'static [&[&'static str]] = &[&["test"], &["other one"]]; // ERROR Consider removing 'static
   |                             -^^^^^^^---- help: consider removing `'static`: `&str`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:14:17
   |
14 | const VAR_SIX: &'static u8 = &5;
   |                -^^^^^^^--- help: consider removing `'static`: `&u8`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:16:29
   |
16 | const VAR_SEVEN: &[&(&str, &'static [&'static str])] = &[&("one", &["other one"])];
   |                            -^^^^^^^--------------- help: consider removing `'static`: `&[&'static str]`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:16:39
   |
16 | const VAR_SEVEN: &[&(&str, &'static [&'static str])] = &[&("one", &["other one"])];
   |                                      -^^^^^^^---- help: consider removing `'static`: `&str`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:18:20
   |
18 | const VAR_HEIGHT: &'static Foo = &Foo {};
   |                   -^^^^^^^---- help: consider removing `'static`: `&Foo`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:20:19
   |
20 | const VAR_SLICE: &'static [u8] = b"Test constant #1"; // ERROR Consider removing 'static.
   |                  -^^^^^^^----- help: consider removing `'static`: `&[u8]`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:22:19
   |
22 | const VAR_TUPLE: &'static (u8, u8) = &(1, 2); // ERROR Consider removing 'static.
   |                  -^^^^^^^--------- help: consider removing `'static`: `&(u8, u8)`

error: Constants have by default a `'static` lifetime
  --> $DIR/const_static_lifetime.rs:24:19
   |
24 | const VAR_ARRAY: &'static [u8; 1] = b"T"; // ERROR Consider removing 'static.
   |                  -^^^^^^^-------- help: consider removing `'static`: `&[u8; 1]`

error: aborting due to 13 previous errors

